vb_accumulator/src/universal.rs

```diff
@@ -133,10 +133,49 @@ impl<E> UniversalAccumulator<E>
 where
     E: PairingEngine,
 {
-    /// Create a new universal accumulator. Given the max size, it generates (max_size+1) initial elements
-    /// and adds them to the accumulator; these initial elements can never be removed or added back to the
-    /// accumulator
+    /// Create a new universal accumulator. Given the max size, it generates `max_size-n+1` initial elements
+    /// as mentioned in the paper. `n` elements are passed as argument `xs`. These are generated
+    /// once for each curve as they depend on the order of the scalar field and then reused.
+    /// Constants for some of the curves are defined in `universal_init_constants.rs` and are computed
+    /// using sage code in `universal_init.sage`. Those should be passed as argument `xs`. The rest of
+    /// the `max_size + 1` are generated randomly.
     pub fn initialize<R: RngCore>(
+        rng: &mut R,
+        setup_params: &SetupParams<E>,
+        max_size: u64,
+        sk: &SecretKey<E::Fr>,
+        xs: Vec<E::Fr>,
+        initial_elements_store: &mut dyn InitialElementsStore<E::Fr>,
+    ) -> Self {
+        let mut f_V = E::Fr::one();
+        for x in xs {
+            f_V = f_V * (x + sk.0);
+            initial_elements_store.add(x);
+        }
+
+        // At least `max_size + 1` initial elements need to be secret. Its assumed that elements in `xs`
```

Comment:
Okay sorry I will be a mathematical pedant here to prevent possible confusion. We need more secret elements than known elements (in case all users collude), so we might need more than max_size+1 secret elements if we initialise with some public values.

Also "Its assumed" should be "It's assumed"

Then we are good to go

Reply:
Updated.